util/metrics/metrics.go

```diff
@@ -0,0 +1,89 @@
+package metrics
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// RegisterPrometheusMetrics register all prometheus metrics with the global
+// metrics handler.
+func RegisterPrometheusMetrics() {
+	prometheus.Register(BlockImportTimeSeconds)
+	prometheus.Register(CumulativeImportTime)
+	prometheus.Register(CumulativeTxns)
+	prometheus.Register(ImportedTxnsPerBlock)
+	prometheus.Register(CurrentRoundGauge)
+	prometheus.Register(CumulativeBlockUploadTime)
+	prometheus.Register(BlockUploadTime)
+}
+
+// Prometheus metric names broken out for reuse.
+const (
+	BlockImportTimeName           = "average_import_time_sec"
+	CumulativeImportTimeName      = "cumulative_import_time_sec"
+	BlockUploadTimeName           = "average_block_upload_time_sec"
+	CumulativeBlockUploadTimeName = "cumulative_block_upload_time_sec"
+	ImportedTxnsPerBlockName      = "average_imported_tx_per_block"
+	CumulativeTxnsName            = "cumulative_imported_tx"
+	CurrentRoundGaugeName         = "current_round"
+)
+
+// AllMetricNames is a reference for all the custom metric names.
+var AllMetricNames = []string{
+	BlockImportTimeName,
+	CumulativeImportTimeName,
+	BlockUploadTimeName,
+	CumulativeBlockUploadTimeName,
+	ImportedTxnsPerBlockName,
+	CumulativeTxnsName,
+	CurrentRoundGaugeName,
+}
+
+// Initialize the prometheus objects.
+var (
+	BlockImportTimeSeconds = prometheus.NewSummary(
+		prometheus.SummaryOpts{
+			Subsystem: "indexer_daemon",
+			Name:      BlockImportTimeName,
+			Help:      "Total block upload and processing time in seconds.",
+		})
+
+	CumulativeImportTime = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Subsystem: "indexer_daemon",
+			Name:      CumulativeImportTimeName,
+			Help:      "Total time in seconds spent uploading and processing blocks.",
+		})
+
+	BlockUploadTime = prometheus.NewSummary(
```

Comment:
Its a subset of the total import + processing time. So if it takes 8 seconds to it import a round with 20k transactions this might tell us that 1.5 seconds we're spent uploading the raw data